vendor/incremental-cycles/src/incremental_cycles.ml

```diff
@@ -78,29 +54,24 @@ let rec visit_backward
   match stack with
   | [] -> VisitBackwardCompleted
   | vertex :: stack ->
-    let res = interruptible_fold (fun y (stack, fuel) ->
-      if fuel = 0 then
-        (* There is no fuel left *)
-        Break true
-      else if is_marked g y mark then
-        (* This vertex has already been visited, skip it *)
-        Continue (stack, fuel - 1)
-      else if vertex_eq y target then
-        (* A path to [target] has been found *)
-        Break false
-      else begin
-        set_mark g y mark;
-        set_parent g y vertex;
-        Continue (y :: stack, fuel - 1)
-      end
-    ) (get_incoming g vertex) (stack, fuel)
-    in
-    match res with
-    | Break timeout ->
-      if timeout then VisitBackwardInterrupted
-      else (set_parent g target vertex; VisitBackwardCyclic)
-    | Continue (stack, fuel) ->
-      visit_backward g target mark fuel stack
+    visit_backward_aux g target mark vertex stack fuel (get_incoming g vertex)
+and visit_backward_aux g target mark vertex stack fuel = function
+  | [] -> visit_backward g target mark fuel stack
+  | y :: ys ->
+    if fuel = 0 then
+      (* There is no fuel left *)
+      VisitBackwardInterrupted
+    else if is_marked g y mark then
+      (* This vertex has already been visited, skip it *)
+      visit_backward_aux g target mark vertex stack fuel ys
```

Comment:
Why is not `fuel` decreased here anymore?

Reply:
indeed it should be.